main.go

```diff
@@ -138,26 +180,63 @@ func main() {
 
 	ctx := ctrl.SetupSignalHandler()
 
-	tlsOptions, err := common.GetSspTlsOptions(ctx)
+	tlsCfgFunc := func(cfg *tls.Config) {
+		// This callback executes on each client call returning a new config to be used
+		// please be aware that the APIServer is using http keepalive so this is going to
+		// be executed only after a while for fresh connections and not on existing ones
+		cfg.GetConfigForClient = func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+			tlsOptions, err := common.GetSspTlsOptions(ctx)
+			if err != nil {
+				setupLog.Error(err, "Error while getting tls profile")
+				os.Exit(1)
+			}
+
+			if tlsOptions.IsEmpty() {
+				// if ciphersSuites is nil then default ciphers will be used
+				cfg.CipherSuites = nil
+				return cfg, nil
+			}
+			cfg.CipherSuites = tlsOptions.CipherIDs(&setupLog)
+			cfg.MinVersion, err = minTLSVersionId(tlsOptions.MinTLSVersion)
+			if err != nil {
+				return nil, err
+			}
+			setupLog.Info("Configured ciphers", "ciphers", cfg.CipherSuites)
+			return cfg, nil
+		}
+	}
+	tlsConfigCallback := []func(*tls.Config){tlsCfgFunc}
+
+	err = runPrometheusServer(metricsAddr, ctx)
 	if err != nil {
-		setupLog.Error(err, "Error while getting tls profile")
+		setupLog.Error(err, "unable to start prometheus server")
 		os.Exit(1)
 	}
 
-	err = runPrometheusServer(metricsAddr, *tlsOptions)
+	tlsOptions, err := common.GetSspTlsOptions(ctx)
 	if err != nil {
-		setupLog.Error(err, "unable to start prometheus server")
+		setupLog.Error(err, "Error while getting tls Profile")
 		os.Exit(1)
 	}
 
+	var webhookServer *webhook.Server
+	if tlsOptions.IsEmpty() {
```

Comment:
Not sure we can do this. I don't think this allows to change the config later without restart?